### ISSUE TITLE
Add prop to disable ordering by a column

### ISSUE THREAD

**Issue by dextermb** [langleyfoxall/react-dynamic-data-table-laravel-api#3]
When displaying appended attributes from a Model in the DataTable and clicking on a custom attribute leads to a `500` response as the attribute doesn't exist in the Model's database table.

##### `User`

```php
class User extends Model {
  protected $appends = [
    'name'
  ];

  public function getNameAttribute()
  {
    return $this->first_name.' '.$this->last_name;
  }
}

```

##### `AjaxDataTable`

```jsx
<AjaxDataTable
  disableFieldSort={[
    'name'
  ]}
/>
```

In the example above `name` is a column made up of multiple _actual_ columns.

**Comment by dextermb** [comment]
This could be achieved on the backend by having some sort of standard method naming pattern (similar to how `getNameAttribute` works), perhaps something along the lines of:

```php
public static function getNameOrderLogic(QueryBuilder $query)
{
  // order by concat first_name and last_name
}
```

It would always search for this logic when clicking on a field. If the method does not exist then it'll use default `orderBy` logic, otherwise it'll use the method.

As suggested in the original post, it should still be on the frontend to give the most amount of options possible.

**Comment by dextermb** [comment]
@DivineOmega I'd still like to see this implemented. Perhaps I should tackle it at some point?

**Comment by Jord-JD** [comment]
Since this PR is on the Laravel API repo, this issue should focus on discussion of the server side part of this (your second comment).

I like the idea of providing alternative sorting logic, but I'd prefer to avoid a syntax based purely on a naming convention. If we can avoid 'magic', I think we should.

Perhaps something similar to the following.

```php
return (new DataTableResponder(User::class, $request))
    ->overrideOrderByLogic([
        'name' => function(Builder $query, string $direction) {
            $query->orderByRaw('CONCAT(first_name, last_name) '.$direction);
        },
    ]);
```

What do you think?

**Comment by Jord-JD** [comment]
Or alternatively:

```php
return (new DataTableResponder(User::class, $request))
    ->overrideOrderByLogic(function(Builder $query, string $field, string $direction) {
        switch($field) {
            case 'name':
                $query->orderByRaw('CONCAT(first_name, last_name) '.$direction);
                break;
        }
    });
```

**Comment by dextermb** [comment]
@DivineOmega I like the solutions offered here, but I think we could also get the responder to optionally automatically determine if a `get[]Attribute` method exists and if it does then disable ordering by it.

```php
return (new DataTableResponder(User::class, $request))
    ->disableOrderingByCustomAttributes();
```

**Comment by Jord-JD** [comment]
@dextermb That would make sense. How would you go about disabling the actual controls client side with this? Perhaps it could pass back down an array of `disableOrderingFields` as part of the response, which the Ajax data table component could read in and adjust itself appropriately?

**Comment by Jord-JD** [comment]
Though I think if this is implemented, perhaps it should be on by default. It'd be a breaking change, but I feel it's a sensible default.

**Comment by dextermb** [comment]
> @dextermb That would make sense. How would you go about disabling the actual controls client side with this? Perhaps it could pass back down an array of `disableOrderingFields` as part of the response, which the Ajax data table component could read in and adjust itself appropriately?

`ApiResponse` on a successful response has `meta`, here could be a list of `orderable_fields` or similar?

**Comment by Jord-JD** [comment]
That could work.

**Comment by dextermb** [comment]
Implemented in #4.
